src/game.cpp

```diff
@@ -915,7 +915,7 @@ void game::create_factions()
 {
     faction tmp;
     std::vector<std::string> faction_vector = tmp.all_json_factions();
-    for(std::vector<std::string>::reverse_iterator it = faction_vector.rbegin(); it != faction_vector.rend(); ++it) {
+    for(auto it = faction_vector.rbegin(); it != faction_vector.rend(); ++it) {
```

Comment:
Don't routinely replace types with auto, there was nothing unclear about this.

Reply:
Similarly this was a case where the actual message referred to cases where what you could gain was a small bit rather than a large one. Yes, you could use std::vector<std::string>::reverse_iterator, but the only thing it does is make it harder to update if we ever want to change from say, a vector to something else. I can remove the two or so cases where that shows up if you'd prefer.

Reply:
It also makes the code harder to read since it hides information.